### PR TITLE
Add XCI rom format support

### PR DESCRIPTION
**Pull request by sspacelynx**

### CODE REVIEW

File: app/src/main/cpp/skyline/loader/xci.cpp [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                        programNca = std::move(nca);
+                    else if (nca.contentType == vfs::NcaContentType::Control && nca.romFs != nullptr)
+                        controlNca = std::move(nca);
+                } catch (const loader_exception &e) {
+                    throw loader_exception(e.error);
+                } catch (const std::exception &e) {
+                    continue;
+                }
+            }
+        } else throw exception("Corrupted secure partition");
```
Comment (bylaws) [inline app/src/main/cpp/skyline/loader/xci.cpp:51]: Newline after else

File: app/src/main/cpp/skyline/loader/xci.cpp [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        }
+
+        if (icon == nullptr)
+            return std::vector<u8>();
+
+        std::vector<u8> buffer(icon->size);
+        icon->Read(buffer);
+        return buffer;
+    }
+
```
Comment (bylaws) [inline app/src/main/cpp/skyline/loader/xci.cpp:88]: Drop this space

**Review by bylaws — CHANGES_REQUESTED** [review]

File: app/src/main/cpp/skyline/loader/xci.cpp (hunk 2 above)
Comment (sspacelynx) [reply app/src/main/cpp/skyline/loader/xci.cpp:88]: oops didn't see this

**Comment by sspacelynx** [comment]
Fixed up changes as the commit was too small to justify

**Review by PixelyIon — CHANGES_REQUESTED** [review]
A single change, that's it

File: app/src/main/cpp/skyline/loader/xci.cpp [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                    else if (nca.contentType == vfs::NcaContentType::Control && nca.romFs != nullptr)
+                        controlNca = std::move(nca);
+                } catch (const loader_exception &e) {
+                    throw loader_exception(e.error);
+                } catch (const std::exception &e) {
+                    continue;
+                }
+            }
+        } else
+            throw exception("Corrupted secure partition");
```
[the remark this answers is not in the document]
Comment (sspacelynx) [reply app/src/main/cpp/skyline/loader/xci.cpp:52]: Missed that

**Comment by sspacelynx** [comment]
Done!

**Review by PixelyIon — APPROVED** [review]
Great job! This is exactly how we wanted it to be implemented :star:

**Review by bylaws — APPROVED** [review]
Good

**Pull request merged**